tinygrad/tensor.py

```diff
@@ -156,9 +156,27 @@ def empty(*shape, **kwargs):
   @staticmethod
   def manual_seed(seed=0): Tensor._seed = seed
 
+  _rng_counter: int = 0
```

Comment:
from what i read, this will need to be device specific. 

how i was planning to do it in my PR was:
`_rng_counter = defaultdict(int)` and then hash device like `device_id = int(hashlib.sha256('CPU'.encode()).hexdigest(), 16) & 0xFFFFFFFF` to get the current counter for the device

also i think below the `0x0` in `ks` should be the `device_id` 

that way the key/counter contain:
- seed
- device id
- device specific counter

which should allow parallel use without getting the same numbers twice